src/app/basket.service.ts

```diff
@@ -368,7 +371,7 @@ export class BasketService {
             let alreadyExists = false;
             object[container].forEach((child) => {
                 let matchedCount = 0;
-                keyNames.forEach((k, v) => {
+                keyNames.forEach((v, k) => {
```

Comment:
`k` refers to the index in the array, should it be called `i`? (very minor)

Reply:
Good point, I always found it weird that it's reversed from every other language..